qiskit_experiments/curve_analysis/scatter_table.py

```diff
@@ -9,185 +9,498 @@
 # Any modifications or derivative works of this code must retain this
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
+
 """Table representation of the x, y data for curve fitting."""
+from __future__ import annotations
 
 import logging
-from typing import List, Sequence, Dict, Any, Union
+import warnings
+from collections.abc import Iterator
+from typing import Any
+from itertools import groupby
+from operator import itemgetter
 
 import numpy as np
 import pandas as pd
 
 from qiskit.utils import deprecate_func
 
-from qiskit_experiments.framework.table_mixin import DefaultColumnsMixIn
-
 
 LOG = logging.getLogger(__name__)
 
 
-class ScatterTable(pd.DataFrame, DefaultColumnsMixIn):
-    """A table to store x and y data with metadata associated with the data point.
+class ScatterTable:
+    """A table-like dataset for curve fitting intermediate data.
+
+    Default table columns are defined in the class attribute :attr:`.DEFAULT_COLUMNS`.
+    This table cannot be expanded with user-provided column names.
+
+    This dataset is not thread safe. Do not use the same instance in multiple threads.
+
+    .. _filter_scatter_table:
+
+    Filtering ScatterTable
```

Comment:
Fair enough. I wrote the docs here just for visibility (since developer may not check website tutorial) but this makes rendered doc bit ugly. I moved this section to the tutorial and added the cross ref link here. Done in [9e27f16](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1360/commits/9e27f16eb3d0f376cc0bf81494319199d67f4cb1)